test/framework/framework.go

```diff
@@ -286,9 +288,28 @@ func (f *Framework) SetDatacenterStatusReady(ctx context.Context, key ClusterKey
 			Status:             corev1.ConditionTrue,
 			LastTransitionTime: now,
 		})
+		dc.Status.ObservedGeneration = dc.Generation
 	})
 }
 
+// UpdateDatacenterGeneration fetches the CassandraDatacenter specified by key and persists
+// a status update to make the ObservedGeneration match the Generation.
+// It will wait for .meta.Generation to be different from .status.ObservedGeneration.
+// This is done to simulate the behavior of the CassandraDatacenter controller.
+func (f *Framework) UpdateDatacenterGeneration(ctx context.Context, t *testing.T, key ClusterKey) bool {
+	dc := &cassdcapi.CassandraDatacenter{}
+	err := f.Get(ctx, key, dc)
+	require.NoError(t, err, "failed to get CassandraDatacenter %s/%s", key.Namespace, key.Name)
+	if dc.Generation == dc.Status.ObservedGeneration {
```

Comment:
This `if` block is unnecessary. The CassandraDatacenter controller isn't running, so nothing (other than tests) should be updating `.Status.ObservedGeneration`. I first added a log statement at line 304 and ran the tests. As expected the log statement was never executed. Then I commented out the block and the tests still passed. Lastly, there isn't any need to wrap the method call in an `Eventually` block.

Reply:
This check has been useful to ensure we're performing the update only if it is necessary. It prevented me from putting generation updates where they didn't belong. That allows making sure we're not complexifying our tests with additional code lines that aren't required.